ERROR: replication should connect HA port in thread_pool occurred in ShardingSphere-Scaling openGauss

## Bug Report

### Which version of ShardingSphere did you use?

master - 49953efd394c2a56d3851fc53c0b8791777bc4f7

### Expected behavior

Scaling succeeded.

### Actual behavior

Logs:
```
[INFO ] 2022-05-11 15:33:47.622 [Connection-1-ThreadExecutor] o.a.s.m.m.c.c.r.c.s.ScalingRegistrySubscriber - start scaling job, locked the schema name, event=MetadataVersionPreparedEvent(version=1, databaseName=bmsql_sharding)
[INFO ] 2022-05-11 15:33:47.622 [Connection-1-ThreadExecutor] o.a.s.d.p.s.r.RuleAlteredJobWorker - Start scaling job by StartScalingEvent(databaseName=bmsql_sharding, activeVersion=0, newVersion=1)
[INFO ] 2022-05-11 15:33:47.662 [Connection-1-ThreadExecutor] o.a.s.d.p.s.r.RuleAlteredJobWorker - type=org.apache.shardingsphere.sharding.yaml.config.YamlShardingRuleConfiguration, ruleAlteredLogicTables=[bmsql_warehouse]
[INFO ] 2022-05-11 15:33:47.687 [Connection-1-ThreadExecutor] o.a.s.d.p.c.a.i.RuleAlteredJobAPIImpl - Start scaling job by RuleAlteredJobConfiguration{jobId='0130317c30317c3054317c626d73716c5f7368617264696e67', databaseName='bmsql_sharding', activeVersion=0, newVersion=1, shardingSize=10000000, sourceDatabaseType='openGauss', targetDatabaseType='openGauss'}
[INFO ] 2022-05-11 15:33:47.709 [Curator-SafeNotifyService-0] o.a.s.d.p.c.e.PipelineJobExecutor - ADDED job config: /scaling/0130317c30317c3054317c626d73716c5f7368617264696e67/config
[INFO ] 2022-05-11 15:33:47.770 [Curator-SafeNotifyService-0] o.a.s.d.p.c.lock.PipelineSimpleLock - tryLock, lockName=bmsql_sharding, timeoutMills=1000, result=true
[INFO ] 2022-05-11 15:33:47.772 [Curator-SafeNotifyService-0] o.a.s.d.p.c.e.PipelineJobExecutor - 0130317c30317c3054317c626d73716c5f7368617264696e67 added to executing jobs success
[INFO ] 2022-05-11 15:33:47.781 [Curator-SafeNotifyService-0] org.quartz.impl.StdSchedulerFactory - Using default implementation for ThreadExecutor
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] o.quartz.core.SchedulerSignalerImpl - Initialized Scheduler Signaller of type: class org.quartz.core.SchedulerSignalerImpl
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] org.quartz.core.QuartzScheduler - Quartz Scheduler v.2.3.2 created.
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] o.a.s.e.l.i.s.JobShutdownHookPlugin - Registering Quartz shutdown hook. 0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] org.quartz.simpl.RAMJobStore - RAMJobStore initialized.
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] org.quartz.core.QuartzScheduler - Scheduler meta-data: Quartz Scheduler (v2.3.2) '0130317c30317c3054317c626d73716c5f7368617264696e67' with instanceId 'NON_CLUSTERED'
  Scheduler class: 'org.quartz.core.QuartzScheduler' - running locally.
  NOT STARTED.
  Currently in standby mode.
  Number of jobs executed: 0
  Using thread pool 'org.quartz.simpl.SimpleThreadPool' - with 1 threads.
  Using job-store 'org.quartz.simpl.RAMJobStore' - which does not support persistence. and is not clustered.

[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] org.quartz.impl.StdSchedulerFactory - Quartz scheduler '0130317c30317c3054317c626d73716c5f7368617264696e67' initialized from an externally provided properties instance.
[INFO ] 2022-05-11 15:33:47.782 [Curator-SafeNotifyService-0] org.quartz.impl.StdSchedulerFactory - Quartz scheduler version: 2.3.2
[INFO ] 2022-05-11 15:33:47.839 [Curator-SafeNotifyService-0] org.quartz.core.QuartzScheduler - Scheduler 0130317c30317c3054317c626d73716c5f7368617264696e67_$_NON_CLUSTERED started.
[INFO ] 2022-05-11 15:33:47.878 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.s.r.RuleAlteredJob - Execute job 0130317c30317c3054317c626d73716c5f7368617264696e67-1
[INFO ] 2022-05-11 15:33:47.878 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.s.r.RuleAlteredJob - Execute job 0130317c30317c3054317c626d73716c5f7368617264696e67-0
[INFO ] 2022-05-11 15:33:47.909 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.a.c.TableNameSchemaNameMapping - mapping={bmsql_warehouse_0=public, bmsql_customer_0=public, bmsql_district=public, bmsql_oorder=public, bmsql_stock=public, gs_errors=dbe_pldeveloper, bmsql_customer=public, bmsql_item=public, bmsql_new_order=public, bmsql_history=public, bmsql_warehouse=public, bmsql_order_line=public, gs_source=dbe_pldeveloper, bmsql_config=public}
[INFO ] 2022-05-11 15:33:47.909 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.a.c.TableNameSchemaNameMapping - mapping={bmsql_warehouse_0=public, bmsql_customer_0=public, bmsql_district=public, bmsql_oorder=public, bmsql_stock=public, gs_errors=dbe_pldeveloper, bmsql_customer=public, bmsql_item=public, bmsql_new_order=public, bmsql_history=public, bmsql_warehouse=public, bmsql_order_line=public, gs_source=dbe_pldeveloper, bmsql_config=public}
[INFO ] 2022-05-11 15:33:47.916 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.a.c.TableNameSchemaNameMapping - mapping={bmsql_warehouse_0=public, bmsql_customer_0=public, bmsql_district=public, bmsql_oorder=public, bmsql_stock=public, gs_errors=dbe_pldeveloper, bmsql_customer=public, bmsql_item=public, bmsql_new_order=public, bmsql_history=public, bmsql_warehouse=public, bmsql_order_line=public, gs_source=dbe_pldeveloper, bmsql_config=public}
[INFO ] 2022-05-11 15:33:47.917 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.s.s.ShardingRuleAlteredJobConfigurationPreparer - createTaskConfiguration, dataSourceName=ds_0, result=TaskConfiguration(jobConfig=RuleAlteredJobConfiguration{jobId='0130317c30317c3054317c626d73716c5f7368617264696e67', databaseName='bmsql_sharding', activeVersion=0, newVersion=1, shardingSize=10000000, sourceDatabaseType='openGauss', targetDatabaseType='openGauss'}, dumperConfig=DumperConfiguration(databaseName=bmsql_sharding, dataSourceName=ds_0, position=null, tableNameMap={bmsql_warehouse=bmsql_warehouse}, tableNameSchemaNameMapping=org.apache.shardingsphere.data.pipeline.api.config.TableNameSchemaNameMapping@342e0b25), importerConfig=ImporterConfiguration(shardingColumnsMap={bmsql_warehouse=[w_id]}, tableNameSchemaNameMapping=org.apache.shardingsphere.data.pipeline.api.config.TableNameSchemaNameMapping@731bd21e, batchSize=1000, retryTimes=3))
[INFO ] 2022-05-11 15:33:47.919 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.a.c.TableNameSchemaNameMapping - mapping={bmsql_warehouse_0=public, bmsql_customer_0=public, bmsql_district=public, bmsql_oorder=public, bmsql_stock=public, gs_errors=dbe_pldeveloper, bmsql_customer=public, bmsql_item=public, bmsql_new_order=public, bmsql_history=public, bmsql_warehouse=public, bmsql_order_line=public, gs_source=dbe_pldeveloper, bmsql_config=public}
[INFO ] 2022-05-11 15:33:47.919 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.s.s.ShardingRuleAlteredJobConfigurationPreparer - createTaskConfiguration, dataSourceName=ds_1, result=TaskConfiguration(jobConfig=RuleAlteredJobConfiguration{jobId='0130317c30317c3054317c626d73716c5f7368617264696e67', databaseName='bmsql_sharding', activeVersion=0, newVersion=1, shardingSize=10000000, sourceDatabaseType='openGauss', targetDatabaseType='openGauss'}, dumperConfig=DumperConfiguration(databaseName=bmsql_sharding, dataSourceName=ds_1, position=null, tableNameMap={bmsql_warehouse=bmsql_warehouse}, tableNameSchemaNameMapping=org.apache.shardingsphere.data.pipeline.api.config.TableNameSchemaNameMapping@4576fce1), importerConfig=ImporterConfiguration(shardingColumnsMap={bmsql_warehouse=[w_id]}, tableNameSchemaNameMapping=org.apache.shardingsphere.data.pipeline.api.config.TableNameSchemaNameMapping@4d743a89, batchSize=1000, retryTimes=3))
[INFO ] 2022-05-11 15:33:48.003 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.c.p.d.AbstractDataSourcePreparer - prepareTargetSchemas, schemaNames=[public], defaultSchema=public
[WARN ] 2022-05-11 15:33:48.495 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] c.zaxxer.hikari.pool.ProxyConnection - HikariPool-11 - Connection org.opengauss.jdbc.PgConnection@59db9331 marked as broken because of SQLSTATE(0A000), ErrorCode(0)
java.sql.SQLFeatureNotSupportedException: Method org.opengauss.jdbc.PgDatabaseMetaData.getRowIdLifetime() is not yet implemented.
	at org.opengauss.Driver.notImplemented(Driver.java:907)
	at org.opengauss.jdbc.PgDatabaseMetaData.getRowIdLifetime(PgDatabaseMetaData.java:2587)
	at com.zaxxer.hikari.pool.HikariProxyDatabaseMetaData.getRowIdLifetime(HikariProxyDatabaseMetaData.java)
	at org.apache.shardingsphere.infra.metadata.resource.CachedDatabaseMetaData.getRowIdLifetimeFromOriginMetaData(CachedDatabaseMetaData.java:437)
	at org.apache.shardingsphere.infra.metadata.resource.CachedDatabaseMetaData.<init>(CachedDatabaseMetaData.java:431)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.createCachedDatabaseMetaData(ShardingSphereMetaData.java:84)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.createResource(ShardingSphereMetaData.java:75)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.create(ShardingSphereMetaData.java:67)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsBuilder.getMetaDataMap(MetaDataContextsBuilder.java:129)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsBuilder.build(MetaDataContextsBuilder.java:116)
	at org.apache.shardingsphere.mode.manager.memory.MemoryContextManagerBuilder.build(MemoryContextManagerBuilder.java:58)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:81)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:64)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:77)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:138)
	at org.apache.shardingsphere.driver.config.datasource.ShardingSpherePipelineDataSourceCreator.createPipelineDataSource(ShardingSpherePipelineDataSourceCreator.java:43)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceFactory.newInstance(PipelineDataSourceFactory.java:44)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceManager.getDataSource(PipelineDataSourceManager.java:52)
	at org.apache.shardingsphere.data.pipeline.core.prepare.datasource.AbstractDataSourcePreparer.getTargetCachedDataSource(AbstractDataSourcePreparer.java:111)
	at org.apache.shardingsphere.data.pipeline.core.prepare.datasource.AbstractDataSourcePreparer.prepareTargetSchemas(AbstractDataSourcePreparer.java:77)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobPreparer.prepareTarget(RuleAlteredJobPreparer.java:138)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobPreparer.prepareAndCheckTarget(RuleAlteredJobPreparer.java:121)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobPreparer.prepareAndCheckTargetWithLock(RuleAlteredJobPreparer.java:101)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobPreparer.prepare(RuleAlteredJobPreparer.java:81)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJob.execute(RuleAlteredJob.java:51)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:33)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:29)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.process(ElasticJobExecutor.java:172)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.lambda$process$0(ElasticJobExecutor.java:153)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
[INFO ] 2022-05-11 15:33:48.558 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.o.p.d.OpenGaussDataSourcePreparer - queryTableDefinition, sql=SELECT * FROM pg_get_tabledef('public.bmsql_warehouse'::regclass::oid)
[INFO ] 2022-05-11 15:33:48.560 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.o.p.d.OpenGaussDataSourcePreparer - getActualTableDefinitions, schemaName=public, dataNode=DataNode(dataSourceName=ds_0, tableName=bmsql_warehouse, schemaName=null), tableDefinition=SET search_path = public;
CREATE TABLE bmsql_warehouse (
    w_id integer NOT NULL,
    w_ytd numeric(12,2),
    w_tax numeric(4,4),
    w_name character varying(10),
    w_street_1 character varying(20),
    w_street_2 character varying(20),
    w_city character varying(20),
    w_state character(2),
    w_zip character(9)
)
WITH (orientation=row, compression=no);
ALTER TABLE bmsql_warehouse ADD CONSTRAINT bmsql_warehouse_pkey_bmsql_warehouse PRIMARY KEY (w_id);
[INFO ] 2022-05-11 15:33:48.564 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.c.p.d.AbstractDataSourcePreparer - execute target table sql: 
CREATE TABLE IF NOT EXISTS public.bmsql_warehouse (
    w_id integer NOT NULL,
    w_ytd numeric(12,2),
    w_tax numeric(4,4),
    w_name character varying(10),
    w_street_1 character varying(20),
    w_street_2 character varying(20),
    w_city character varying(20),
    w_state character(2),
    w_zip character(9)
)

[INFO ] 2022-05-11 15:33:48.934 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.c.p.d.AbstractDataSourcePreparer - execute target table sql: 
ALTER TABLE public.bmsql_warehouse ADD CONSTRAINT bmsql_warehouse_pkey_bmsql_warehouse PRIMARY KEY (w_id)
[INFO ] 2022-05-11 15:33:48.951 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.o.p.d.OpenGaussDataSourcePreparer - create target table 'bmsql_warehouse' success
[INFO ] 2022-05-11 15:33:48.952 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.c.c.d.AbstractDataSourceChecker - checkEmpty, sql=SELECT * FROM public.bmsql_warehouse LIMIT 1
[INFO ] 2022-05-11 15:33:49.085 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.c.m.l.PipelineTableMetaDataLoader - loadTableMetaData, schemaNameFinal=public, tableNamePattern=bmsql_warehouse, result={bmsql_warehouse=PipelineTableMetaData(name=bmsql_warehouse, columnMetaDataMap={w_id=PipelineColumnMetaData(ordinalPosition=1, name=w_id, dataType=4, dataTypeName=int4, primaryKey=true), w_ytd=PipelineColumnMetaData(ordinalPosition=2, name=w_ytd, dataType=2, dataTypeName=numeric, primaryKey=false), w_tax=PipelineColumnMetaData(ordinalPosition=3, name=w_tax, dataType=2, dataTypeName=numeric, primaryKey=false), w_name=PipelineColumnMetaData(ordinalPosition=4, name=w_name, dataType=12, dataTypeName=varchar, primaryKey=false), w_street_1=PipelineColumnMetaData(ordinalPosition=5, name=w_street_1, dataType=12, dataTypeName=varchar, primaryKey=false), w_street_2=PipelineColumnMetaData(ordinalPosition=6, name=w_street_2, dataType=12, dataTypeName=varchar, primaryKey=false), w_city=PipelineColumnMetaData(ordinalPosition=7, name=w_city, dataType=12, dataTypeName=varchar, primaryKey=false), w_state=PipelineColumnMetaData(ordinalPosition=8, name=w_state, dataType=1, dataTypeName=bpchar, primaryKey=false), w_zip=PipelineColumnMetaData(ordinalPosition=9, name=w_zip, dataType=1, dataTypeName=bpchar, primaryKey=false)}, columnNames=[w_id, w_ytd, w_tax, w_name, w_street_1, w_street_2, w_city, w_state, w_zip], primaryKeyColumns=[w_id])}, cost time=43 ms
[INFO ] 2022-05-11 15:33:49.088 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.s.r.p.InventoryTaskSplitter - getPositionByPrimaryKeyRange, endId is 0, break, tableName=bmsql_warehouse, primaryKey=w_id, beginId=6
[INFO ] 2022-05-11 15:33:49.091 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.s.r.RuleAlteredJobPreparer - prepare, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67, shardingItem=1, inventoryTasks=[InventoryTask(taskId=ds_1.bmsql_warehouse#0, position=i,0,5)], incrementalTasks=[IncrementalTask(taskId=ds_1)]
[INFO ] 2022-05-11 15:33:49.113 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-2] o.a.s.d.p.s.r.RuleAlteredJobSchedulerCenter - start RuleAlteredJobScheduler, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67, shardingItem=1
[INFO ] 2022-05-11 15:33:49.115 [Thread-8] o.a.s.d.p.s.r.RuleAlteredJobScheduler - -------------- Start inventory task --------------
[INFO ] 2022-05-11 15:33:49.116 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.InventoryTask@3f687ef7
[INFO ] 2022-05-11 15:33:49.117 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@125ae2d5
[INFO ] 2022-05-11 15:33:49.117 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:49.118 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLInventoryDumper@70928d51
[INFO ] 2022-05-11 15:33:49.119 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump, uniqueKeyDataType=4, firstSQL=SELECT * FROM public.bmsql_warehouse WHERE w_id >= ? AND w_id <= ? ORDER BY w_id ASC LIMIT ?, laterSQL=SELECT * FROM public.bmsql_warehouse WHERE w_id > ? AND w_id <= ? ORDER BY w_id ASC LIMIT ?, position=i,0,5
[INFO ] 2022-05-11 15:33:49.131 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump done, round=3, maxUniqueKeyValue=Optional.empty
[INFO ] 2022-05-11 15:33:49.131 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump, before put FinishedRecord
[INFO ] 2022-05-11 15:33:52.173 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - write, get FinishedRecord, break
[INFO ] 2022-05-11 15:33:52.174 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=4, finishedByBreak=true
[INFO ] 2022-05-11 15:33:52.175 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.core.task.InventoryTask - importer future done
[INFO ] 2022-05-11 15:33:52.176 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.s.r.RuleAlteredJobScheduler - onSuccess, all inventory tasks finished.
[INFO ] 2022-05-11 15:33:52.176 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.core.task.InventoryTask - importer onSuccess, taskId=ds_1.bmsql_warehouse#0
[INFO ] 2022-05-11 15:33:52.176 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.s.r.RuleAlteredJobScheduler - -------------- Start incremental task --------------
[INFO ] 2022-05-11 15:33:52.177 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask@6d9b3d2e
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@758860f2
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@792c4bc
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@227a8ff4
[INFO ] 2022-05-11 15:33:52.178 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:52.182 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper@1c7e5b33
[ERROR] 2022-05-11 15:33:52.194 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - Incremental task execute failed.
org.apache.shardingsphere.data.pipeline.core.ingest.exception.IngestException: org.opengauss.util.PSQLException: [127.0.0.1:38870/127.0.0.1:25433] ERROR: replication should connect HA port in thread_pool
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.dump(OpenGaussWalDumper.java:93)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.doStart(OpenGaussWalDumper.java:74)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask.doStart(IncrementalTask.java:85)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.run(AbstractLifecycleExecutor.java:61)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.opengauss.util.PSQLException: [127.0.0.1:38870/127.0.0.1:25433] ERROR: replication should connect HA port in thread_pool
	at org.opengauss.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:603)
	at org.opengauss.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:164)
	at org.opengauss.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
	at org.opengauss.core.ConnectionFactory.openConnection(ConnectionFactory.java:53)
	at org.opengauss.jdbc.PgConnection.<init>(PgConnection.java:255)
	at org.opengauss.Driver.makeConnection(Driver.java:561)
	at org.opengauss.Driver.connect(Driver.java:314)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:190)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.wal.OpenGaussLogicalReplication.createConnection(OpenGaussLogicalReplication.java:55)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.getReplicationConnectionUnwrap(OpenGaussWalDumper.java:106)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.dump(OpenGaussWalDumper.java:79)
	... 12 common frames omitted
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop job 0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop inventory task 0130317c30317c3054317c626d73716c5f7368617264696e67 - ds_1.bmsql_warehouse#0
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.InventoryTask@3f687ef7
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLInventoryDumper@70928d51
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@125ae2d5
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop incremental task 0130317c30317c3054317c626d73716c5f7368617264696e67 - ds_1
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask@6d9b3d2e
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper@1c7e5b33
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@758860f2
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@792c4bc
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@227a8ff4
[INFO ] 2022-05-11 15:33:52.195 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobContext - close...
[INFO ] 2022-05-11 15:33:52.200 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobWorker - restoreSourceWriting, databaseName=bmsql_sharding
[INFO ] 2022-05-11 15:33:52.200 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobWorker - Source writing is still stopped on database 'bmsql_sharding', restore it now
[ERROR] 2022-05-11 15:33:52.201 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobWorker - restore source writing failed, databaseName=bmsql_sharding
org.apache.shardingsphere.mode.repository.cluster.ClusterPersistRepositoryException: java.lang.IllegalStateException: Not acquired
	at org.apache.shardingsphere.mode.repository.cluster.zookeeper.handler.CuratorZookeeperExceptionHandler.handleException(CuratorZookeeperExceptionHandler.java:53)
	at org.apache.shardingsphere.mode.repository.cluster.zookeeper.lock.ZookeeperInternalLockHolder$ZookeeperInternalLock.unlock(ZookeeperInternalLockHolder.java:117)
	at org.apache.shardingsphere.mode.manager.cluster.coordinator.lock.global.GlobalLockRegistryService.releaseLock(GlobalLockRegistryService.java:46)
	at org.apache.shardingsphere.mode.manager.cluster.coordinator.lock.global.ShardingSphereDistributeGlobalLock.releaseLock(ShardingSphereDistributeGlobalLock.java:139)
	at org.apache.shardingsphere.mode.manager.cluster.coordinator.lock.global.general.ShardingSphereGeneralLock.releaseLock(ShardingSphereGeneralLock.java:75)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobWorker.restoreSourceWriting(RuleAlteredJobWorker.java:328)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobWorker.scalingReleaseDatabaseLevelLock(RuleAlteredJobWorker.java:313)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
	at com.google.common.eventbus.EventBus.post(EventBus.java:212)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobScheduler$2.onFailure(RuleAlteredJobScheduler.java:142)
	at org.apache.shardingsphere.data.pipeline.core.execute.ExecuteEngine$ExecuteFutureCallback.onFailure(ExecuteEngine.java:120)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1066)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Not acquired
	at org.apache.curator.shaded.com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex.release(InterProcessSemaphoreMutex.java:75)
	at org.apache.shardingsphere.mode.repository.cluster.zookeeper.lock.ZookeeperInternalLockHolder$ZookeeperInternalLock.unlock(ZookeeperInternalLockHolder.java:113)
	... 22 common frames omitted
[INFO ] 2022-05-11 15:33:52.201 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.c.lock.PipelineSimpleLock - releaseLock, lockName=bmsql_sharding
[INFO ] 2022-05-11 15:33:53.003 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.s.r.RuleAlteredJobPreparer - unlocked, lockName=prepare-0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-11 15:33:53.066 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.c.m.l.PipelineTableMetaDataLoader - loadTableMetaData, schemaNameFinal=public, tableNamePattern=bmsql_warehouse, result={bmsql_warehouse=PipelineTableMetaData(name=bmsql_warehouse, columnMetaDataMap={w_id=PipelineColumnMetaData(ordinalPosition=1, name=w_id, dataType=4, dataTypeName=int4, primaryKey=true), w_ytd=PipelineColumnMetaData(ordinalPosition=2, name=w_ytd, dataType=2, dataTypeName=numeric, primaryKey=false), w_tax=PipelineColumnMetaData(ordinalPosition=3, name=w_tax, dataType=2, dataTypeName=numeric, primaryKey=false), w_name=PipelineColumnMetaData(ordinalPosition=4, name=w_name, dataType=12, dataTypeName=varchar, primaryKey=false), w_street_1=PipelineColumnMetaData(ordinalPosition=5, name=w_street_1, dataType=12, dataTypeName=varchar, primaryKey=false), w_street_2=PipelineColumnMetaData(ordinalPosition=6, name=w_street_2, dataType=12, dataTypeName=varchar, primaryKey=false), w_city=PipelineColumnMetaData(ordinalPosition=7, name=w_city, dataType=12, dataTypeName=varchar, primaryKey=false), w_state=PipelineColumnMetaData(ordinalPosition=8, name=w_state, dataType=1, dataTypeName=bpchar, primaryKey=false), w_zip=PipelineColumnMetaData(ordinalPosition=9, name=w_zip, dataType=1, dataTypeName=bpchar, primaryKey=false)}, columnNames=[w_id, w_ytd, w_tax, w_name, w_street_1, w_street_2, w_city, w_state, w_zip], primaryKeyColumns=[w_id])}, cost time=45 ms
[INFO ] 2022-05-11 15:33:53.068 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.s.r.p.InventoryTaskSplitter - getPositionByPrimaryKeyRange, endId is 0, break, tableName=bmsql_warehouse, primaryKey=w_id, beginId=7
[INFO ] 2022-05-11 15:33:53.068 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.s.r.RuleAlteredJobPreparer - prepare, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67, shardingItem=0, inventoryTasks=[InventoryTask(taskId=ds_0.bmsql_warehouse#0, position=i,0,6)], incrementalTasks=[IncrementalTask(taskId=ds_0)]
[INFO ] 2022-05-11 15:33:53.072 [elasticjob-0130317c30317c3054317c626d73716c5f7368617264696e67-1] o.a.s.d.p.s.r.RuleAlteredJobSchedulerCenter - start RuleAlteredJobScheduler, jobId=0130317c30317c3054317c626d73716c5f7368617264696e67, shardingItem=0
[INFO ] 2022-05-11 15:33:53.073 [Thread-9] o.a.s.d.p.s.r.RuleAlteredJobScheduler - -------------- Start inventory task --------------
[INFO ] 2022-05-11 15:33:53.074 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.InventoryTask@5457a875
[INFO ] 2022-05-11 15:33:53.075 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLInventoryDumper@5eafc305
[INFO ] 2022-05-11 15:33:53.076 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@4db1ebd9
[INFO ] 2022-05-11 15:33:53.076 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:53.076 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump, uniqueKeyDataType=4, firstSQL=SELECT * FROM public.bmsql_warehouse WHERE w_id >= ? AND w_id <= ? ORDER BY w_id ASC LIMIT ?, laterSQL=SELECT * FROM public.bmsql_warehouse WHERE w_id > ? AND w_id <= ? ORDER BY w_id ASC LIMIT ?, position=i,0,6
[INFO ] 2022-05-11 15:33:53.084 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump done, round=3, maxUniqueKeyValue=Optional.empty
[INFO ] 2022-05-11 15:33:53.085 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.d.AbstractInventoryDumper - inventory dump, before put FinishedRecord
[INFO ] 2022-05-11 15:33:55.184 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:55.184 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:55.185 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:55.188 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.core.task.IncrementalTask - importer onSuccess, taskId=ds_1
[WARN ] 2022-05-11 15:33:56.555 [ShardingSphere-Scaling-execute-0] c.zaxxer.hikari.pool.ProxyConnection - HikariPool-17 - Connection org.opengauss.jdbc.PgConnection@2903d4b0 marked as broken because of SQLSTATE(0A000), ErrorCode(0)
java.sql.SQLFeatureNotSupportedException: Method org.opengauss.jdbc.PgDatabaseMetaData.getRowIdLifetime() is not yet implemented.
	at org.opengauss.Driver.notImplemented(Driver.java:907)
	at org.opengauss.jdbc.PgDatabaseMetaData.getRowIdLifetime(PgDatabaseMetaData.java:2587)
	at com.zaxxer.hikari.pool.HikariProxyDatabaseMetaData.getRowIdLifetime(HikariProxyDatabaseMetaData.java)
	at org.apache.shardingsphere.infra.metadata.resource.CachedDatabaseMetaData.getRowIdLifetimeFromOriginMetaData(CachedDatabaseMetaData.java:437)
	at org.apache.shardingsphere.infra.metadata.resource.CachedDatabaseMetaData.<init>(CachedDatabaseMetaData.java:431)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.createCachedDatabaseMetaData(ShardingSphereMetaData.java:84)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.createResource(ShardingSphereMetaData.java:75)
	at org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData.create(ShardingSphereMetaData.java:67)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsBuilder.getMetaDataMap(MetaDataContextsBuilder.java:129)
	at org.apache.shardingsphere.mode.metadata.MetaDataContextsBuilder.build(MetaDataContextsBuilder.java:116)
	at org.apache.shardingsphere.mode.manager.memory.MemoryContextManagerBuilder.build(MemoryContextManagerBuilder.java:58)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:81)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:64)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:77)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:138)
	at org.apache.shardingsphere.driver.config.datasource.ShardingSpherePipelineDataSourceCreator.createPipelineDataSource(ShardingSpherePipelineDataSourceCreator.java:43)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceFactory.newInstance(PipelineDataSourceFactory.java:44)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceManager.getDataSource(PipelineDataSourceManager.java:52)
	at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.write(AbstractImporter.java:88)
	at org.apache.shardingsphere.data.pipeline.core.importer.AbstractImporter.doStart(AbstractImporter.java:74)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.run(AbstractLifecycleExecutor.java:61)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
[INFO ] 2022-05-11 15:33:56.591 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - write, get FinishedRecord, break
[INFO ] 2022-05-11 15:33:56.591 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=4, finishedByBreak=true
[INFO ] 2022-05-11 15:33:56.591 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.core.task.InventoryTask - importer future done
[INFO ] 2022-05-11 15:33:56.592 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.core.task.InventoryTask - importer onSuccess, taskId=ds_0.bmsql_warehouse#0
[INFO ] 2022-05-11 15:33:56.592 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.s.r.RuleAlteredJobScheduler - onSuccess, all inventory tasks finished.
[INFO ] 2022-05-11 15:33:56.593 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.s.r.RuleAlteredJobScheduler - -------------- Start incremental task --------------
[INFO ] 2022-05-11 15:33:56.594 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask@6ac8c47b
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@54f264b3
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@604f98d8
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-0] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper@9f5dea8
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.c.i.AbstractImporter - importer write
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.a.e.AbstractLifecycleExecutor - start lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@349137a
[INFO ] 2022-05-11 15:33:56.595 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.c.i.AbstractImporter - importer write
[ERROR] 2022-05-11 15:33:56.603 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - Incremental task execute failed.
org.apache.shardingsphere.data.pipeline.core.ingest.exception.IngestException: org.opengauss.util.PSQLException: [127.0.0.1:38880/127.0.0.1:25433] ERROR: replication should connect HA port in thread_pool
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.dump(OpenGaussWalDumper.java:93)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.doStart(OpenGaussWalDumper.java:74)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask.doStart(IncrementalTask.java:85)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.run(AbstractLifecycleExecutor.java:61)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.opengauss.util.PSQLException: [127.0.0.1:38880/127.0.0.1:25433] ERROR: replication should connect HA port in thread_pool
	at org.opengauss.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:603)
	at org.opengauss.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:164)
	at org.opengauss.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
	at org.opengauss.core.ConnectionFactory.openConnection(ConnectionFactory.java:53)
	at org.opengauss.jdbc.PgConnection.<init>(PgConnection.java:255)
	at org.opengauss.Driver.makeConnection(Driver.java:561)
	at org.opengauss.Driver.connect(Driver.java:314)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:190)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.wal.OpenGaussLogicalReplication.createConnection(OpenGaussLogicalReplication.java:55)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.getReplicationConnectionUnwrap(OpenGaussWalDumper.java:106)
	at org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper.dump(OpenGaussWalDumper.java:79)
	... 12 common frames omitted
[INFO ] 2022-05-11 15:33:56.603 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop job 0130317c30317c3054317c626d73716c5f7368617264696e67
[INFO ] 2022-05-11 15:33:56.603 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop inventory task 0130317c30317c3054317c626d73716c5f7368617264696e67 - ds_0.bmsql_warehouse#0
[INFO ] 2022-05-11 15:33:56.603 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.InventoryTask@5457a875
[INFO ] 2022-05-11 15:33:56.603 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLInventoryDumper@5eafc305
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@4db1ebd9
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - stop incremental task 0130317c30317c3054317c626d73716c5f7368617264696e67 - ds_0
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask@6ac8c47b
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.ingest.OpenGaussWalDumper@9f5dea8
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@54f264b3
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@604f98d8
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.a.e.AbstractLifecycleExecutor - stop lifecycle executor: org.apache.shardingsphere.data.pipeline.opengauss.importer.OpenGaussImporter@349137a
[INFO ] 2022-05-11 15:33:56.604 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobContext - close...
[INFO ] 2022-05-11 15:33:56.608 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.s.r.RuleAlteredJobWorker - restoreSourceWriting, databaseName=bmsql_sharding
[INFO ] 2022-05-11 15:33:56.609 [ShardingSphere-Scaling-execute-4] o.a.s.d.p.c.lock.PipelineSimpleLock - releaseLock, lockName=bmsql_sharding
[INFO ] 2022-05-11 15:33:59.602 [ShardingSphere-Scaling-execute-1] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:59.602 [ShardingSphere-Scaling-execute-2] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:59.602 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.c.i.AbstractImporter - importer write done, rowCount=0, finishedByBreak=false
[INFO ] 2022-05-11 15:33:59.602 [ShardingSphere-Scaling-execute-3] o.a.s.d.p.core.task.IncrementalTask - importer onSuccess, taskId=ds_0

```

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

Using openGauss 3.0 and the following configurations:

```
omm=# show wal_sender_timeout;
 wal_sender_timeout 
--------------------
 0
(1 row)

omm=# show wal_level;
 wal_level 
-----------
 logical
(1 row)

```

ShardingSphere-Proxy using driver `org.opengauss:opengauss-jdbc:3.0.0`
Using Cluster mode

```yaml
schemaName: bmsql_sharding
dataSources:
  ds_0:
    url: jdbc:opengauss://127.0.0.1:25433/bmsql_0?loggerLevel=OFF
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0
  ds_1:
    url: jdbc:opengauss://127.0.0.1:25433/bmsql_1?loggerLevel=OFF
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0

rules:
  - !SHARDING
    defaultDatabaseStrategy:
      none:
    defaultTableStrategy:
      none:
    keyGenerators:
      snowflake:
        props:
          worker-id: 123
        type: SNOWFLAKE
    tables:
      bmsql_warehouse:
        actualDataNodes: ds_${0..1}.bmsql_warehouse
        databaseStrategy:
          standard:
            shardingColumn: w_id
            shardingAlgorithmName: mod

    shardingAlgorithms:
      mod:
        type: MOD
        props:
          sharding-count: 2

    scalingName: bmsql_scaling
    scaling:
      scaling_name1:
        completionDetector:
          props:
            incremental-task-idle-seconds-threshold: '60'
          type: IDLE
        dataConsistencyChecker:
          props:
            chunk-size: '1000'
          type: DATA_MATCH
```

Add data sources and alter sharding rule:
```sql
ADD RESOURCE target_0 (
    URL="jdbc:opengauss://127.0.0.1:35433/bmsql_0",
    USER=omm,
    PASSWORD=123,
    PROPERTIES("maximumPoolSize"=64, "minimumIdle"=0,"idleTimeout"="60000")
), target_1 (
    URL="jdbc:opengauss://127.0.0.1:35433/bmsql_1",
    USER=omm,
    PASSWORD=123,
    PROPERTIES("maximumPoolSize"=64, "minimumIdle"=0,"idleTimeout"="60000")
), target_2 (
    URL="jdbc:opengauss://127.0.0.1:35433/bmsql_2",
    USER=omm,
    PASSWORD=123,
    PROPERTIES("maximumPoolSize"=64, "minimumIdle"=0,"idleTimeout"="60000")
);

ALTER SHARDING TABLE RULE bmsql_warehouse(
    RESOURCES(target_0, target_1, target_2),
    SHARDING_COLUMN=w_id,
    TYPE(NAME=MOD,PROPERTIES("sharding-count"=3))
);
```

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
@@ -21,6 +21,9 @@ import lombok.extern.slf4j.Slf4j;
 import org.apache.shardingsphere.data.pipeline.api.datasource.config.impl.StandardPipelineDataSourceConfiguration;
 import org.apache.shardingsphere.data.pipeline.api.datasource.config.yaml.YamlJdbcConfiguration;
 import org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.BaseLogSequenceNumber;
+import org.apache.shardingsphere.infra.database.metadata.url.JdbcUrl;
+import org.apache.shardingsphere.infra.database.metadata.url.StandardJdbcUrlParser;
+import org.apache.shardingsphere.infra.database.type.dialect.OpenGaussDatabaseType;
 import org.opengauss.PGProperty;
 import org.opengauss.jdbc.PgConnection;
 import org.opengauss.replication.LogSequenceNumber;
@@ -37,6 +40,8 @@ import java.util.Properties;
 @Slf4j
 public final class OpenGaussLogicalReplication {
     
+    private static final String HA_PORT_ERROR_MESSAGE_KEY = "HA port";
+    
     /**
      * Create connection.
      *
@@ -52,7 +57,24 @@ public final class OpenGaussLogicalReplication {
         PGProperty.ASSUME_MIN_SERVER_VERSION.set(props, "9.4");
         PGProperty.REPLICATION.set(props, "database");
         PGProperty.PREFER_QUERY_MODE.set(props, "simple");
-        return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
+        try {
+            return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
+        } catch (final SQLException ex) {
+            if (ex.getMessage().contains(HA_PORT_ERROR_MESSAGE_KEY)) {
+                log.info("Failed to connect to openGauss caused by: {} - {}. Try connecting to HA port.", ex.getSQLState(), ex.getMessage());
+                return tryConnectingToHAPort(jdbcConfig.getJdbcUrl(), props);
+            }
+            throw ex;
+        }
+    }
+    
+    private Connection tryConnectingToHAPort(final String jdbcUrl, final Properties props) throws SQLException {
+        JdbcUrl parseResult = new StandardJdbcUrlParser().parse(jdbcUrl);
+        PGProperty.PG_HOST.set(props, parseResult.getHostname());
+        PGProperty.PG_DBNAME.set(props, parseResult.getDatabase());
+        int haPort = parseResult.getPort() + 1;
+        PGProperty.PG_PORT.set(props, haPort);
+        return DriverManager.getConnection(new OpenGaussDatabaseType().getJdbcUrlPrefixes().iterator().next(), props);
     }
     
     /**

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
@@ -60,7 +60,7 @@ public final class OpenGaussLogicalReplication {
         try {
             return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
         } catch (final SQLException ex) {
-            if (failedBecauseOfHAPort(ex)) {
+            if (failedBecauseOfNonHAPort(ex)) {
                 log.info("Failed to connect to openGauss caused by: {} - {}. Try connecting to HA port.", ex.getSQLState(), ex.getMessage());
                 return tryConnectingToHAPort(jdbcConfig.getJdbcUrl(), props);
             }
@@ -68,7 +68,7 @@ public final class OpenGaussLogicalReplication {
         }
     }
     
-    private boolean failedBecauseOfHAPort(final SQLException ex) {
+    private boolean failedBecauseOfNonHAPort(final SQLException ex) {
         return ex.getMessage().contains(HA_PORT_ERROR_MESSAGE_KEY);
     }
     

--- a/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
+++ b/shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/OpenGaussLogicalReplication.java
@@ -60,12 +60,16 @@ public final class OpenGaussLogicalReplication {
         try {
             return DriverManager.getConnection(jdbcConfig.getJdbcUrl(), props);
         } catch (final SQLException ex) {
-            if (ex.getMessage().contains(HA_PORT_ERROR_MESSAGE_KEY)) {
+            if (failedBecauseOfHAPort(ex)) {
                 log.info("Failed to connect to openGauss caused by: {} - {}. Try connecting to HA port.", ex.getSQLState(), ex.getMessage());
                 return tryConnectingToHAPort(jdbcConfig.getJdbcUrl(), props);
             }
             throw ex;
         }
+    }
+    
+    private boolean failedBecauseOfHAPort(final SQLException ex) {
+        return ex.getMessage().contains(HA_PORT_ERROR_MESSAGE_KEY);
     }
     
     private Connection tryConnectingToHAPort(final String jdbcUrl, final Properties props) throws SQLException {

